### PR TITLE
Remove libopenmpt port

### LINKED ISSUE #18480: libopenmpt port has been in abysmal shape from the start
@atkawa7 @alexkaratarakis @ras0219-msft @sagamusix

Even though I (libopenmpt upstream maintainer) have been initially positive about seeing libopenmpt packaged in vcpkg, I sadly have now changed my opinion after seeing this port completely unmaintained for 3 years. This is *REALLY VERY BAD*.

As time has shown, vcpkg is unable to properly maintain ports it contains. The port of libopenmpt included in vcpkg was broken from the start and did get merged in without any proper review. My comments on the original pull request (#2669) (which I only ever noticed after it was already merged) were never addressed. In particular, it has wrong dependencies, advertises a version number that breaks our SemVer versioning scheme, is a development and not a released version, contains multiple security issues by now [1] (as it never got updated), builds components it should not build, and other issues. Noone should use that port or should have used it, EVER.

The state of libopenmpt in vcpkg shines a very bad light on our project as well as on vcpkg itself. Seeing how bad our own software gets maintained as a vcpkg port, I will assume all other ports are of equally bad quality. Thus I will avoid to ever use vcpkg. To each and everyone asking, I will also strongly recommend to avoid vcpkg completely. vcpkg has proven itself to be unable or unwilling to deal with even basic software development best practices. vcpkg is shipping broken software. There is not even a recommended way to report security issues as far as I can see.

I am really frustrated here. You are packaging our software in a completely broken fashion, and do not even care to fix it.

Please either do your absolutely basic duties of at least maintaining the libopenmpt port properly, or just remove it completely.

[1]
 *  https://lib.openmpt.org/libopenmpt/2018/02/03/security-update-0.3.6/ (CVE-2018-6611)
 *  https://lib.openmpt.org/libopenmpt/2018/04/08/security-updates-0.3.8-0.2-beta31-0.2.7561-beta20.5-p8-0.2.7386-beta20.3-p11/ (CVE-2018-10017)
 *  https://lib.openmpt.org/libopenmpt/2018/04/29/security-updates-0.3.9-0.2-beta32-0.2.7561-beta20.5-p9-0.2.7386-beta20.3-p12/ (CVE-2018-11710)
 *  https://lib.openmpt.org/libopenmpt/2018/07/28/security-updates-0.3.11-0.2.10635-beta34-0.2.7561-beta20.5-p10-0.2.7386-beta20.3-p13/ (CVE-2018-20861)
 *  https://lib.openmpt.org/libopenmpt/2018/10/21/security-updates-0.3.13-0.2.10933-beta36-0.2.7561-beta20.5-p11-0.2.7386-beta20.3-p14/ (CVE-2018-20860)
 *  https://lib.openmpt.org/libopenmpt/2019/01/22/security-updates-0.4.2-0.3.15-0.2.11253-beta37-0.2.7561-beta20.5-p13-0.2.7386-beta20.3-p16/ (CVE-2019-14382, CVE-2019-14383)
 *  https://lib.openmpt.org/libopenmpt/2019/02/11/security-update-0.4.3/ (CVE-2019-14381) (maybe, as the commit was taken from master which branched to 0.4 later, I will not dig this out any further)
 *  https://lib.openmpt.org/libopenmpt/2019/05/27/security-update-0.4.5/ (CVE-2019-14380) (maybe, again)
 *  https://lib.openmpt.org/libopenmpt/2019/09/30/security-update-0.4.8/ (maybe, again)
 *  https://lib.openmpt.org/libopenmpt/2021/04/11/security-updates-0.5.8-0.4.20-0.3.29/

### PR DESCRIPTION
**Pull request by alexkaratarakis**
Given the issues raised by the official maintainers, remove this
port until a satisfactory port is submitted.

Fixes #18480

### CODE REVIEW

**Review by JackBoosY — APPROVED** [review]

File: toolsrc/VERSION.txt [hunk 1]
```diff
@@ -1 +0,0 @@
-"9999.99.99"
```
Comment (JackBoosY) [inline toolsrc/VERSION.txt:1]: @BillyONeal Should we remove this?

**Comment by Hoikas** [comment]
Why not fix the port to work correctly?

**Comment by BillyONeal** [comment]
> Why not fix the port to work correctly?

It seems like the upstream author's wishes ( in #18480 ) disagree with being packaged in vcpkg at all? Many of the things they don't like are design rules for ports (e.g. being accessible from cmake in a consistent form on all platforms).

I'm weakly in favor of outright removal in compliance to the upstream maintainer's wishes

File: toolsrc/VERSION.txt (hunk 1 above)
Comment (alexkaratarakis) [reply toolsrc/VERSION.txt:1]: That shouldn't be removed. Fixing.

**Review by JackBoosY — CHANGES_REQUESTED** [review]
Also, we should remove the baseline record:
https://github.com/microsoft/vcpkg/blob/6e024e744e7717c06ddacd5089401109c6298553/versions/baseline.json#L3403-L3406

And _https://github.com/microsoft/vcpkg/blob/master/versions/l-/libopenmpt.json_.

**Comment by alexkaratarakis** [comment]
Done.

**Review by JackBoosY — APPROVED** [review]

**Comment by manxorist** [comment]
> > Why not fix the port to work correctly?

> It seems like the upstream author's wishes ( in #18480 ) disagree with being packaged in vcpkg at all?

No, I object to the completely unmaintained state that the libopenmpt has been in from the beginning. If vcpkg is unable to properly update its ports and/or does not even fix valid concerns that I have risen 3 years ago, a port is of no use to anyone,

> Many of the things they don't like are design rules for ports (e.g. being accessible from cmake in a consistent form on all platforms).

I did not disagree with supporting cmake, and I even considered taking upstream contribution (https://github.com/microsoft/vcpkg/pull/2669#discussion_r164781796), but that never happened.

If vcpkg did not require signing an utterly unacceptable and legally questionable CLA, I would happily contribute a cmake build system for libopenmpt in vcpkg myself, however in the current licensing state this will certainly not happen.

**Comment by Hoikas** [comment]
While I am not a user of this port, it *was* accepted into vcpkg. That means that clearly, at some point, someone was and *still may be using* it. As a user in general, I would be very disappointed if an entire port I depended on were deleted in response to a bug report. Even if that bug report is from the maintainer. Within this mindset, I think it is inappropriate to "throw the baby out with the bathwater" and simply delete the port. Further, deleting something when a bug report comes in requesting an improvement seems like an unprofessional, knee-jerk response, IMO. It would be better for the port to be fixed to use the native buildsystem(s) - plenty of other ports have to deal with upstream build systems of dire quality (eg python, openssl), and this will just be another one.

**Comment by ras0219-msft** [comment]
Closing in favor of resolving https://github.com/microsoft/vcpkg/issues/18480 via another way.

Nice to see you again @alexkaratarakis :)

**Pull request closed without merging**